src/Hico/Config/Cli.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-# LANGUAGE DuplicateRecordFields #-}
+
+module Hico.Config.Cli where
+
+import           Data.Semigroup ((<>))
+import           Data.Maybe
+import           Hico.Types
+import           Options.Applicative
+import           SDL (RendererType(SoftwareRenderer), defaultRenderer, rendererType)
+
+data CliConfig = CliConfig {
+  widthIn  :: Int,
+  heightIn :: Int,
+  renderer :: Maybe RendererType
+}
+
+processRunConfig :: CliConfig -> GameConfig
+processRunConfig raw = GameConfig{
+   width    = widthIn raw
+   ,height  = heightIn raw
+  ,renderer = fromMaybe defaultRendererType (rendererRaw raw)
+  }
+  where
+    rendererRaw raw = renderer(raw:: CliConfig)
+
+xresP :: Parser Int
```

Comment:
These options need to be thought about carefully. The API (at least atm) relies heavily on knowing the width and height of the game world. For instance, the rect and text commands etc, all rely on absolute pixel values. So if you built a game against one resolution, then tried to run it with different config params, it would not work. 

One way to do it, would be to scale those absolute values.

Reply:
That sounds reasonable to do for now, I'll proceed along those lines, thanks!

Reply:
We may need to tweak the wiring of what I push to actually have it make sense, at some point, as I'm still working off the black screen version of the example game.